tests/test_favicon.py

```diff
@@ -201,3 +201,16 @@ def test_request_kwargs(m):
li class="menu-home menu-item-770"><a href="index.php">Home</a></li><li class="menu-about menu-item-16"><a href="http://mock.com/about">About</a></li><li class="menu-shop sf-dropdown menu-item-27"><a>Shop</a><ul class="sf-dropdown-menu">	<li class="menu-pacific-rim menu-item-29"><a target="_blank" href="http://pacificrim.mock.com">Pacific Rim</a></li>	<li class="menu-the-janoskians menu-item-30"><a target="_blank" href="http://janoskians.mock.com">The Janoskians</a></li></ul></li><li class="menu-contact-us menu-item-24"><a href="http://mock.com/contact-us">Contact Us</a></li></ul> </div> </div> </div> </div> <!-- Close Container --> </header> <div class="wrap contentclass" role="document"> <div class="sliderclass kad-desktop-slider"> <div id="imageslider" class="container"> <div class="flexslider kt-flexslider loading" style="max-width:1170px; margin-left: auto; margin-right:auto;" data-flex-speed="7000" data-flex-anim-speed="600" data-flex-animation="fade" data-flex-auto="1"> <ul class="slides"> <li> <a href="http://janoskians.mock.com" target="_self"> <img src="http://mock.com/wp-content/uploads/2015/06/Large_Janoskians2.png" alt="" /> </a> </li> </ul> </div> <!--Flex Slides--> </div><!--Container--></div><!--sliderclass--> <div id="content" class="container homepagecontent"> <div class="row"> <div class="main col-md-12" role="main"> <div class="entry-content" itemprop="mainContentOfPage"> <div class="homecontent clearfix home-margin"> <p><a href="http://pacificrim.mock.com"><img class="aligncenter size-full wp-image-1027" src="http://mock.com/wp-content/uploads/2013/07/Wide-Pacific-Rim.png" alt="Wide-Pacific-Rim" width="1150" height="290" srcset="http://mock.com/wp-content/uploads/2013/07/Wide-Pacific-Rim.png 1150w, http://mock.com/wp-content/uploads/2013/07/Wide-Pacific-Rim-300x76.png 300w, http://mock.com/wp-content/uploads/2013/07/Wide-Pacific-Rim-1024x258.png 1024w" sizes="(max-width: 1150px) 100vw, 1150px" /></a></p> </div> </div> </div><!-- /.main --> </div><!-- /.row--> </div><!-- /.content --> </div><!-- /.wrap --> <footer id="containerfooter" class="footerclass" role="contentinfo"> <div class="container"> <div class="row"> </div> <div class="footercredits clearfix"> <p>&copy; 2020 MOCK.com</p> </div> </div></footer><script type='text/javascript' src='http://mock.com/wp-content/plugins/responsive-image-maps/jquery.rwdImageMaps.min.js?ver=1.5'></script><script type='text/javascript' src='http://mock.com/wp-includes/js/imagesloaded.min.js?ver=3.2.0'></script><script type='text/javascript' src='http://mock.com/wp-includes/js/masonry.min.js?ver=3.3.2'></script><script type='text/javascript' src='http://mock.com/wp-content/themes/virtue/assets/js/min/plugins-min.js?ver=255'></script><script type='text/javascript' src='http://mock.com/wp-content/themes/virtue/assets/js/main.js?ver=255'></script><script type='text/javascript' src='http://mock.com/wp-includes/js/wp-embed.min.js?ver=5.1.1'></script> </div><!--Wrapper--> </body></html>''' # noqa
```

Comment:
Remove unnecessary html that is not related to the test. For example, the icons should only come from `<head>` and not the contents of `<body>`. Keep it simple :)

```
<html>
   <head>
      icons to look for here
   </head>
   <body>
       <p>Test</p>
   </body>
</html>

```

Reply:
While some unit tests test edge cases or specific situation, I wanted this test to simulate real-life as best we can. Anyways, I removed the extra HTML.

Reply:
Done